extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-attachment.js

```diff
@@ -222,5 +225,59 @@ describes.realWin(
       const {activeProductData} = storeService.get(StateProperty.SHOPPING_DATA);
       expect(activeProductData).to.be.null;
     });
+
+    async function setupPDP() {
+      await layoutShoppingImplAndAttachmentChildImpl();
+      // Override to simulate data for one product on the page.
+      storeService.dispatch(Action.ADD_SHOPPING_DATA, {
+        page1: {
+          [shoppingData.productId]: shoppingData.items[0],
+        },
+      });
+      storeService.dispatch(Action.TOGGLE_PAGE_ATTACHMENT_STATE, true);
+    }
+
+    it('should call analytics service on PLP card click', async () => {
+      const trigger = env.sandbox.stub(analytics, 'triggerEvent');
+
+      await layoutShoppingImplAndAttachmentChildImpl();
+      storeService.dispatch(Action.TOGGLE_PAGE_ATTACHMENT_STATE, true);
+      const plpCard = attachmentChildEl.querySelector(
+        '.i-amphtml-amp-story-shopping-plp-card'
+      );
+      plpCard.dispatchEvent(new Event('click'));
+
+      expect(trigger).to.have.been.calledWith(
+        StoryAnalyticsEvent.SHOPPING_PDP_VIEW
+      );
+    });
+
+    it('should call analytics service on buy now button click', async () => {
+      const trigger = env.sandbox.stub(analytics, 'triggerEvent');
+
+      await setupPDP();
+
+      attachmentChildEl
+        .querySelector('.i-amphtml-amp-story-shopping-pdp-cta')
+        .dispatchEvent(new Event('click'));
+
+      expect(trigger).to.have.been.calledWith(
+        StoryAnalyticsEvent.SHOPPING_BUY_NOW_CLICK
+      );
+    });
+
+    it('should call analytics service on attachment state update', async () => {
+      const trigger = env.sandbox.stub(analytics, 'triggerEvent');
+
+      await setupPDP();
+
+      attachmentChildEl
+        .querySelector('.i-amphtml-amp-story-shopping-pdp-cta')
+        .dispatchEvent(new Event('click'));
```

Comment:
Clicks on the "Buy Now" button shouldn't send a PDP_VIEW. 
Opening a PDP in the attachment should.

Reply:
Yes, we don't need to click, it's all handled in the `updateTemplate_` function.